packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,334 @@
+        }
+        return uri;
+    }
+
+    public getEnvironment(_scope?: Construct): {[key: string]: string} | undefined {
+        if (!(this.props)) { return undefined; }
+        return this.props.environment;
+    }
+
+    public getModelDataUrl(_scope?: Construct): string | undefined {
+        if (!(this.props)) { return undefined; }
+        return this.props.modelDataUrl;
+    }
+
+    public getContainerHostname(_scope?: Construct): string | undefined {
+        if (!(this.props)) { return undefined; }
+        return this.props.containerHostname;
+    }
+  }
+
+/**
+ * Construction properties for a SageMaker Model.
+ *
+ * @experimental
+ */
+export interface ModelProps {
+
+    /**
+     * The IAM role that the Amazon SageMaker service assumes.
+     *
+     * @default a new IAM role will be created.
+     */
+    readonly role?: iam.IRole;
+
+    /**
+     * Name of the SageMaker Model.
+     *
+     * @default none
+     */
+    readonly modelName?: string;
+
+    /**
+     * The VPC to deploy the endpoint to.
+     *
+     * @default none
+     */
+    readonly vpc?: ec2.IVpc;
+
+    /**
+     * The VPC subnets to deploy the endpoints.
+     *
+     * @default none
+     */
+    readonly vpcSubnets?: ec2.SubnetSelection;
+
+    /**
+     * Primary container definition
+     *
+     * @default none
+     */
+    readonly primaryContainer?: IContainerDefinition;
+}
+
+/**
+ * Defines a SageMaker Model.
+ *
+ * @experimental
+ */
+export class Model extends Resource implements IModel, ec2.IConnectable {
+
+    /**
+     * Creates a SageMaker model from a name.
+     * @param scope the Construct scope.
+     * @param id the resource id.
+     * @param modelName the name of the model.
+     */
+    public static fromModelName(scope: Construct, id: string, modelName: string): IModel {
+        class Import extends Resource implements IModel {
+          public modelName = modelName;
+          public modelArn = Stack.of(this).formatArn({
+            service: 'sagemaker',
+            resource: 'model',
+            resourceName: this.modelName
+          });
+        }
+
+        return new Import(scope, id);
+      }
+
+    /**
+     * Returns the ARN of this model.
+     * @attribute
+     */
+    public readonly modelArn: string;
+
+    /**
+     * Returns the name of the model.
+     * @attribute
+     */
+    public readonly modelName: string;
+
+    /**
+     * Allows specify security group connections for instances of this fleet.
+     */
+    public readonly connections: ec2.Connections;
+
+    private readonly vpc: ec2.IVpc;
+    private readonly role: iam.IRole;
+    private readonly securityGroup: ec2.ISecurityGroup;
+    private readonly securityGroups: ec2.ISecurityGroup[] = [];
+    private readonly subnets: ec2.SelectedSubnets;
+    private readonly primaryContainer: IContainerDefinition;
+    private readonly containers: IContainerDefinition[] = [];
```

Comment:
Same comment here as in Notebook - everything that's not used outside the constructor should be a local variable instead of a field.